proposals/date_replace.json

```diff
@@ -0,0 +1,108 @@
+{
+    "id": "date_replace",
+    "summary": "Replaces a part of a date",
+    "description": "Replaces a part of a given date string with a new value based on the provided component, allowing for easy replacing of dates.\n\n In case a date/time overflow, the component will be clipped to their respective bounds e.g. if hour is set to 27, it can be clipped to 23. Similarly, if day is set to -40, it can be clipped to the first day of the month",
+    "categories": [
+        "date & time"
+    ],
+    "experimental": true,
+    "parameters": [
+        {
+            "name": "date",
+            "description": "The date (and optionally time) to replace.\n\nIf the given date doesn't include the time, the process assumes that the time component is `00:00:00Z` (i.e. midnight, in UTC). The millisecond part of the time is optional and defaults to `0` if not given.",
+            "schema": [
+                {
+                    "type": "string",
+                    "format": "date-time",
+                    "subtype": "date-time"
+                },
+                {
+                    "type": "string",
+                    "format": "date",
+                    "subtype": "date"
+                }
+            ]
+        },
+        {
+            "name": "value",
+            "description": "The period of time in the component given to replace the existing value.",
+            "schema": {
+                "type": "integer"
+            }
+        },
+        {
+            "name": "component",
```

Comment:
Is there a particular reason to have the "component" parameter after the "value" parameter?

I think it is more intuitive and symmetric to follow this argument order:
- `date_get_component(date, component)`
- `date_replace_component(date, component, value)`